runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -64,6 +70,37 @@ void incrementLapCount(Client client, XpTrackerService xpTrackerService)
 		lapsTillGoal = goalRemainingXp > 0 ? (int) Math.ceil(goalRemainingXp / courseTotalExp) : 0;
 	}
 
+	void calculateLapsPerHour()
+	{
+		Instant now = Instant.now();
```

Comment:
You can simplify this to something like:
```java
		Instant now = Instant.now();

		if (lastLapCompleted != null)
		{
			Duration timeSinceLastLap = Duration.between(lastLapCompleted, now);

			if (!timeSinceLastLap.isNegative())
			{
				lastLapTimes.add(timeSinceLastLap);

				Duration sum = Duration.ofSeconds(0);
				for (Duration lapTime : lastLapTimes)
				{
					sum = sum.plus(lapTime);
				}

				Duration averageLapTime = sum.dividedBy(lastLapTimes.size());
				lapsPerHour = Duration.ofHours(1).toMillis() / averageLapTime.toMillis();
			}
		}

		lastLapCompleted = now;
```

Reply:
That is better, I have fixed this